### PR TITLE
Fix clippy warning

### PR DESCRIPTION
**Pull request by k-nasa**
Looking at GitHub actions, it seems that clippy warnnings are accumulating.
```
Clippy results: 0 ICE, 0 errors, 31 warnings, 0 notes, 0 help
```

### CODE REVIEW

**Comment by skade** [comment]
Especially the `Default` implementations are not implemented in stdlib and we should think about following that lint or turning it off instead.

File: src/task/block_on.rs [hunk 1]
```diff
@@ -69,12 +69,11 @@ where
         let future = task_local::add_finalizer(future);
 
         let future = async move {
```
Comment (skade) [inline src/task/block_on.rs]: For @stjepang: This is not an equivalent transform. Reading the surrounding code, I also don't think that matters, but can you please revalidate?

**Comment by k-nasa** [comment]
@skade I think the default trait should be as much as possible.
The library user should not be saddened by the lack of Default trati.
So I think lint should not be turned off.
How about ❓

File: src/task/block_on.rs (hunk 1 above)
Comment (stjepang) [reply src/task/block_on.rs]: This change looks fine by me. It doesn't really matter because `res` is `()`.

**Comment by k-nasa** [comment]
I don't have the right to see it, but it looks like bors is waiting

bors ping

**Comment by yoshuawuyts** [comment]
Yep, merging manually -- this has been an issue for all PRs for the last day or so.

**Pull request merged**